src/main/java/dk/kb/image/util/ImageAccessValidation.java

```diff
@@ -0,0 +1,144 @@
+package dk.kb.image.util;
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.StreamingOutput;
+
+import dk.kb.license.invoker.v1.ApiException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import dk.kb.image.config.ServiceConfig;
+import dk.kb.license.client.v1.DsLicenseApi;
+import dk.kb.license.model.v1.CheckAccessForIdsInputDto;
+import dk.kb.license.model.v1.CheckAccessForIdsOutputDto;
+import dk.kb.license.model.v1.UserObjAttributeDto;
+import dk.kb.license.util.DsLicenseClient;
+import dk.kb.util.Resolver;
+import dk.kb.util.webservice.exception.InternalServiceException;
+
+public class ImageAccessValidation {
+    private static final Logger log = LoggerFactory.getLogger(ImageAccessValidation.class);
+    private static DsLicenseApi licenseClient;
+    
+    public static enum ACCESS_TYPE {
+        ACCESS, NO_ACCESS, ID_NON_EXISTING
+    };
+
+    
+    
+    public static ACCESS_TYPE accessTypForImage(String resource_id) throws ApiException {
+
+        // Add filter query from license module.
+        DsLicenseApi licenseClient = getDsLicenseApiClient();
+        CheckAccessForIdsInputDto licenseQueryDto = getCheckAccessForIdsInputDto(resource_id);
+        CheckAccessForIdsOutputDto checkAccessForIds = licenseClient.checkAccessForResourceIds(licenseQueryDto); // Use
```

Comment:
We, as service runners, would probably want to know when this fails, so maybe wrap in `try...catch`, log the exception with the `resource_id` and throw an internal server error service exception?

Reply:
Try-catch and log. throw internalserverexception (500)